File: src/backend_dotnet/HRMApi/Models/ApprovalHistory.cs
using System;
using System.Collections.Generic;

namespace HRMApi.Models;

public partial class ApprovalHistory
{
    public int Id { get; set; }

    public int RequestId { get; set; }

    public int ApproverId { get; set; }

    public string Status { get; set; } = null!;

    public string? Note { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual Employee Approver { get; set; } = null!;

    public virtual Request Request { get; set; } = null!;
}

File: src/backend_java/src/main/java/com/group07/human_resource_management/entity/ApprovalHistory.java
package com.group07.human_resource_management.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "approval_history")
@Getter @Setter
@Builder @NoArgsConstructor @AllArgsConstructor
public class ApprovalHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // FK request_id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "request_id", nullable = false)
    private Request request;

    // FK approver_id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "approver_id", nullable = false)
    private Employee approver;

    private String status; // approved / rejected

    private String note;

    @Column(name = "created_at")
    private LocalDateTime createdAt;
}
